MongoDB replica set in MONGODB_URI not supported

In case of a MongoDB replica `,host2[:port2]` is set in MONGODB_URI like it's defined by MongoDB in their [standard URI connection scheme](https://docs.mongodb.com/manual/reference/connection-string/#standard-connection-string-format):
```
mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
```

For example:
```
mongodb://username:password@database-host1:12345,database-host2:12345/database?replicaSet=rs-database
```

This is currently not supported by alerta and will result in this stacktrace:
```
[ERROR] Exception in worker process
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/arbiter.py", line 557, in spawn_worker
    worker.init_process()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/base.py", line 136, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/util.py", line 357, in import_app
    __import__(module)
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/__init__.py", line 96, in <module>
    db = Mongo()
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/database.py", line 30, in __init__
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/database.py", line 48, in connect
    self.connect()
    app.config['MONGO_PORT'] = uri.port
  File "/app/.heroku/python/lib/python2.7/urlparse.py", line 113, in port
    port = int(port, 10)
ValueError: invalid literal for int() with base 10: '12345,database-host2'
```

This is caused by `uri = urlparse(mongo_uri)` in https://github.com/guardian/alerta/blob/master/alerta/app/database.py, because urlparse is not able to parse the scheme with multiple hosts/ports.

Besides this MongoClient is handling this URI perfectly fine.

An alternative would be to use `pymongo.uri_parser`. But, looking a few lines ahead, it's not possible to map multiple hosts to 1 `app.config['MONGO_HOST']` and 1 `app.config['MONGO_PORT']`. Is setting `app.config['XXX']` really necessary there? I've been looking around in the code, but to me it seems those settings are never used after this.

--- a/alerta/app/database.py
+++ b/alerta/app/database.py
@@ -35,58 +35,24 @@ class Mongo(object):
                      os.environ.get('MONGODB_URI', None) or
                      os.environ.get('MONGOHQ_URL', None) or
                      os.environ.get('MONGOLAB_URI', None))
-        if mongo_uri:
-            try:
-                self._client = MongoClient(mongo_uri, connect=False)
-            except Exception as e:
-                LOG.error('MongoDB Client connection error - %s : %s', mongo_uri, e)
-                sys.exit(1)
 
-            uri = urlparse(mongo_uri)
+        if 'MONGO_PORT' in os.environ and 'tcp://' in os.environ['MONGO_PORT']:  # Docker
+            mongo_uri = os.environ['MONGO_PORT'] + os.environ['MONGO_NAME']
 
-            app.config['MONGO_HOST'] = uri.hostname
-            app.config['MONGO_PORT'] = uri.port
-            app.config['MONGO_DATABASE'] = uri.path.lstrip('/')
-            app.config['MONGO_REPLSET'] = None
+        mongo_uri = mongo_uri or app.config['MONGO_URI']  # use app config if no env var overrides
 
-            if uri.username:
-                app.config['MONGO_USERNAME'] = uri.username
-                app.config['MONGO_PASSWORD'] = uri.password
+        try:
+            self._client = MongoClient(mongo_uri, connect=False)
+        except Exception as e:
+            LOG.error('MongoDB Client: ERROR - %s : %s', mongo_uri, e)
+            sys.exit(1)
+        LOG.info('MongoDB Client: Connected to %s', mongo_uri)
 
-        elif 'MONGO_PORT' in os.environ and 'tcp://' in os.environ['MONGO_PORT']:  # Docker
-            host, port = os.environ['MONGO_PORT'][6:].split(':')
-            try:
-                self._client = MongoClient(host, int(port), connect=False)
-            except Exception as e:
-                LOG.error('MongoDB Client connection error - %s : %s', os.environ['MONGO_PORT'], e)
-                sys.exit(1)
-
-        elif not app.config['MONGO_REPLSET']:
-            try:
-                self._client = MongoClient(app.config['MONGO_HOST'], app.config['MONGO_PORT'], connect=False)
-            except Exception as e:
-                LOG.error('MongoDB Client connection error - %s:%s : %s', app.config['MONGO_HOST'], app.config['MONGO_PORT'], e)
-                sys.exit(1)
-            LOG.debug('Connected to mongodb://%s:%s/%s', app.config['MONGO_HOST'], app.config['MONGO_PORT'], app.config['MONGO_DATABASE'])
-
+        if app.config['MONGO_DATABASE']:
+            self._db = self._client[app.config['MONGO_DATABASE']]
         else:
-            try:
-                self._client = MongoClient(app.config['MONGO_HOST'], app.config['MONGO_PORT'], replicaSet=app.config['MONGO_REPLSET'], connect=False)
-            except Exception as e:
-                LOG.error('MongoDB Client ReplicaSet connection error - %s:%s (replicaSet=%s) : %s',
-                          app.config['MONGO_HOST'], app.config['MONGO_PORT'], app.config['MONGO_REPLSET'], e)
-                sys.exit(1)
-            LOG.debug('Connected to mongodb://%s:%s/%s?replicaSet=%s', app.config['MONGO_HOST'],
-                      app.config['MONGO_PORT'], app.config['MONGO_DATABASE'], app.config['MONGO_REPLSET'])
-
-        self._db = self._client[app.config['MONGO_DATABASE']]
-
-        if app.config['MONGO_PASSWORD']:
-            try:
-                self._db.authenticate(app.config['MONGO_USERNAME'], password=app.config['MONGO_PASSWORD'])
-            except Exception as e:
-                LOG.error('MongoDB authentication failed: %s', e)
-                sys.exit(1)
+            self._db = self._client.get_default_database()
+        LOG.info('MongoDB Client: MongoDB v%s, using database "%s"', self.get_version(), self.get_info())
 
         self._create_indexes()
 

--- a/alerta/app/shell.py
+++ b/alerta/app/shell.py
@@ -37,5 +37,4 @@ def main():
     args = parser.parse_args()
 
     LOG.info('Starting alerta version %s ...', __version__)
-    LOG.info('Using MongoDB version %s ...', db.get_version())
     app.run(host=args.host, port=args.port, debug=args.debug, threaded=True)

--- a/alerta/settings.py
+++ b/alerta/settings.py
@@ -18,13 +18,8 @@ QUERY_LIMIT = 10000  # maximum number of alerts returned by a single query
 HISTORY_LIMIT = 100  # cap the number of alert history entries
 
 # MongoDB
-MONGO_HOST = 'localhost'
-MONGO_PORT = 27017
-MONGO_DATABASE = 'monitoring'
-MONGO_REPLSET = None  # 'alerta'
-
-MONGO_USERNAME = 'alerta'
-MONGO_PASSWORD = None
+MONGO_URI = 'mongodb://localhost:27017/monitoring'
+MONGO_DATABASE = 'monitoring'  # can be used to override default database, above
 
 AUTH_REQUIRED = False
 ADMIN_USERS = []

